lib/src/main/java/com/dynatrace/metric/util/Normalize.java

```diff
@@ -144,10 +146,32 @@ static String dimensionValue(String value) {
     value = re_dv_controlCharactersEnd.matcher(value).replaceAll("");
     value = re_dv_controlCharacters.matcher(value).replaceAll("_");
 
+    return value;
+  }
+
+  static String escapeDimensionValue(String val) {
     // escape characters matched by regex with backslash. $1 inserts the matched character.
-    value = re_dv_charactersToEscape.matcher(value).replaceAll("\\\\$1");
+    String escaped = re_dv_charactersToEscape.matcher(val).replaceAll("\\\\$1");
+    if (escaped.length() > dv_max_length) {
+      escaped = escaped.substring(0, dv_max_length);
+      Matcher matcher = re_dv_trailingBackslashes.matcher(escaped);
+      if (matcher.find()) {
+        // string has trailing backslashes. Since every backslash must be escaped, there must be an
+        // even number of backslashes, otherwise the substring operation cut an escaped character
+        // in half: e.g.: "some_long_string\" -> escaped: "some_long_string\\" -> cut with substring
+        // results in "some_long_string\" since the two slashes were on either side of the char
```

Comment:
```suggestion
        // in half: e.g.: "some_long_string," -> escaped: "some_long_string\," -> cut with substring
        // results in "some_long_string\" since the two slashes were on either side of the char
```
This example would be a bit clearer (since it would not result in the same escaped string).